runtime/commonMain/src/kotlinx/serialization/builtins/PrimitiveSerializers.kt

```diff
@@ -99,10 +99,20 @@ public fun Boolean.Companion.serializer(): KSerializer<Boolean> = kotlinx.serial
 public fun BooleanArraySerializer(): KSerializer<BooleanArray> = BooleanArraySerializer
 
 /**
- * Returns serializer for [UnitSerializer] with [descriptor][SerialDescriptor] of [StructureKind.OBJECT] kind.
+ * Returns serializer for [Unit] with [descriptor][SerialDescriptor] of [StructureKind.OBJECT] kind.
  */
+@Deprecated(
+    message = "Deprecated in the favour of Unit.serializer() extension",
+    level = DeprecationLevel.ERROR,
+    replaceWith = ReplaceWith("Unit.serializer()", imports = ["kotlinx.serialization.builtins.serializer"])
+)
 public fun UnitSerializer(): KSerializer<Unit> = kotlinx.serialization.internal.UnitSerializer
```

Comment:
Please move this declaration to Migration.kt to keep them in one pleace

Reply:
There's no Migration.kt in builtins package